### PR TITLE
Merge all setup-* actions into setup-tebako

### PR DESCRIPTION
**Pull request by CAMOBAP**

### CODE REVIEW

**Comment by CAMOBAP** [comment]
To be merged after https://github.com/tamatebako/tebako/pull/124

**Comment by CAMOBAP** [comment]
@maxirmx maybe you can have a quick look at https://github.com/tamatebako/tebako/actions/runs/7777325970/job/21205493548#step:4:989 it looks like i missed something

**Comment by maxirmx** [comment]
> @maxirmx maybe you can have a quick look at https://github.com/tamatebako/tebako/actions/runs/7777325970/job/21205493548#step:4:989 it looks like i missed something

@CAMOBAP 
Sometimes it creates broken cache. I tried to undrestand why it happened but it looked random.
I have deleted cache

Also MacOS runner hangs regulary. I will re-run this job as well.

File: .github/actions/setup-tebako/action.yml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            echo "CXX=${{ inputs.cxx || 'clang++' }}" >> $GITHUB_ENV
+            echo "VCPKG_FORCE_SYSTEM_BINARIES=1" >> $GITHUB_ENV
+            echo "ALPINE_VER=$(cat /etc/alpine-release)" >> $GITHUB_ENV
+        elif [ "$OS" = "Darwin" ]; then
+            echo "CC=${{ inputs.cc || 'clang' }}" >> $GITHUB_ENV
+            echo "CXX=${{ inputs.cxx || 'clang++' }}" >> $GITHUB_ENV
+            echo "XCODE_VERSION=13.1" >> $GITHUB_ENV
+        else
+            echo "Unsupported operating system."
+        fi
```
Comment (maxirmx) [inline .github/actions/setup-tebako/action.yml:97]: I would create "advanced OS" environment variable ("Ubuntu/Alpine/MacOS") on the previous step ("Setup environment") and reuse it.  
Consider the case when we decide to support Fedora as well

File: .github/actions/setup-tebako/action.yml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      if: inputs.cache != 'off'
+      shell: bash
+      run: mkdir -p ${{ inputs.cache_path }}
+
+    - name: Process cache
+      if: inputs.cache != 'off'
+      uses: actions/cache@v4
+      with:
+        path: ${{ inputs.cache_path }}
+        key: ${{ runner.os }}-${{ env.CC }}-v${{ inputs.cache_ver }}
```
Comment (maxirmx) [inline .github/actions/setup-tebako/action.yml:170]: This key means the same cache for Ubuntu and Alpine. That does not work.
Please see the comment above also

**Review by maxirmx — CHANGES_REQUESTED** [review]

File: .github/actions/setup-tebako/action.yml (hunk 1 above)
Comment (CAMOBAP) [reply .github/actions/setup-tebako/action.yml:101]: ADVANCED_OS env has been added

File: .github/actions/setup-tebako/action.yml (hunk 2 above)
Comment (CAMOBAP) [reply .github/actions/setup-tebako/action.yml]: `runner.os` replaced with `env.ADVANCED_OS`

**Comment by maxirmx** [comment]
Something is wrong with MacOS but I suppose it is some brew library update that just happened at the same time.
I will have to check it separately

**Review by maxirmx — APPROVED** [review]
Thank you, @CAMOBAP

**Pull request merged**